drivers/include/lm75.h

```diff
@@ -0,0 +1,166 @@
+
+/* LM75 configuration struct type */
+typedef struct lm75_params {
+    i2c_t   i2c_bus;
+    uint8_t i2c_addr;
+    uint8_t shutdown_mode;
+    uint8_t tm_mode;
+    uint8_t polarity;
+    uint8_t fault_q;
+    uint8_t conv_rate;
+    uint8_t one_shot_mode;
+    uint8_t device_type;
+    lm75_temp_resolutions lm75_res;
+} lm75_params_t;
+
+/* M75 device descriptor */
+typedef struct lm75 {
+    lm75_params_t	lm75_params;
+} lm75_t;
+
+
+/**
+ * @brief Initialization of the LM75 sensor
+ *
+ * Initializes the sensor according to specific input parameters.
+ *
+ * @param[out] dev        device structure to initialize
+ * @param[in] params      initialization parameters
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ * @return LM75_ERROR, on initialization related error
+ */
+int lm75_init(lm75_t *dev, const lm75_params_t *params);
+
+/**
+ * @brief Temperature values of LM75 sensor
+ *
+ * Reads the sensor temperature values from TEMP_REG.
+ *
+ * @param[out] dev                device structure
+ * @param[out] temperature        buffer where temperature value will be written
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ */
+int lm75_get_temperature(lm75_t *dev, int *temperature);
+
+/**
+ * @brief Sets the values for Overtemperature shutdown(OS) and Hysteresis temperature(HYST).
+ * OS gives the temperature's higher bound and HYST the lower bound
+ *
+ * @param[out] dev          device structure
+ * @param[in] temp_os       desired OS temperature in degrees celcius
+ * @param[in] temp_hyst     desired HYST temperature in degrees celcius
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ * @return LM75_ERROR, on temperature setting related error
+ */
+int lm75_set_alarm(lm75_t *dev, int temp_hyst, int temp_os);
+
+/**
+ * @brief Overshutdown temperature value of LM75 sensor
+ *
+ * Reads the sensor OS temperature value from TOS_REG.
+ *
+ * @param[out] dev                device structure
+ * @param[out] temperature        buffer where OS temperature value will be written
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ */
+int lm75_get_os_temp(lm75_t *dev, int *temperature);
+
+/**
+ * @brief Hysteresis temperature value of LM75 sensor
+ *
+ * Reads the sensor hysteresis temperature value from THYST_REG.
+ *
+ * @param[in] dev                 device structure
+ * @param[out] temperature        buffer where HYST temperature value will be written
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ */
+int lm75_get_hyst_temp(lm75_t *dev, int *temperature);
+
+/**
+ * @brief Activate the LM75 sensor shutdown mode
+ *
+ * @param[in] dev      device structure to set into shutdown mode
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR, on mode switching related error
+ */
+int lm75_shutdown(lm75_t *dev);
```

Comment:
```suggestion
int lm75_poweroff(lm75_t *dev);
```

for consistency with other drivers

Reply:
I called that function shutdown because it's also referred extensively in the datasheet as the shutdown mode. Do you think I should leave it like this for more consitency with the datasheet or change it for more consistency with the other drivers?
In my opinion, I think it would be better if it stayed like this, but of course I can change it, if you think it's for the better.

Reply:
A driver might do more than just expose raw hardware functions, here poweron/poweroff are functions common to most devices and they should be named the same everywhere.